### ISSUE TITLE
[RyuJIT/ARM32] Segmentation fault due to regression from recent commit

### ISSUE THREAD

**Issue by hqueue** [dotnet/coreclr#13225]
After merging #13145, several tests fail with Segmentation fault.

This is due to incorrect condition in a helper at https://github.com/dotnet/coreclr/pull/13145/files#diff-93fb9832a03a9783e2a415b581fcb45cR5894 .

At least following 8 tests are failed.
```
FAILED   - [   0][   4s]JIT/Directed/IL/Tailcall/JitTailcall2/JitTailcall2.sh
FAILED   - [   1][   7s]JIT/jit64/hfa/main/testA/hfa_sd2A_d/hfa_sd2A_d.sh
FAILED   - [   2][   8s]JIT/Methodical/fp/exgen/10w250d_cs_do/10w250d_cs_do.sh
FAILED   - [   3][  22s]JIT/Methodical/fp/exgen/10w5d_cs_do/10w5d_cs_do.sh
FAILED   - [   4][  22s]JIT/Methodical/fp/exgen/10w5d_cs_ro/10w5d_cs_ro.sh
FAILED   - [   0][   3s]JIT/Regression/CLR-x86-JIT/V1-M11-Beta1/b48614/b48614/b48614.sh
FAILED   - [   1][   4s]JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b35486/b35486/b35486.sh
FAILED   - [   2][   3s]JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b50027/b50027/b50027.sh
FAILED   - [   3][   3s]JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b69225/b69225/b69225.sh
```

**Comment by hqueue** [comment]
@sjsinju Thank you for notifying me this.  I think you are looking into this helper while updating #13100 too. Can you please fix this along together with #13100 ?

/cc: @dotnet/arm32-contrib

**Comment by sjsinju** [comment]
@hqueue OK. It would be fixed with https://github.com/dotnet/coreclr/pull/13100

**Comment by BruceForstall** [comment]
@sjsinju @hqueue #13100 just got merged; can you verify the fix on these tests, and close the issue?

**Comment by hqueue** [comment]
@BruceForstall Just finished pri2 test and confirmed that this regression is fixed.

Fixed by #13100
